samples/gen-js-wag-patch/index.js

```diff
@@ -0,0 +1,107 @@
+
+const discovery = require("@clever/discovery");
+const request = require("request");
+const url = require("url");
+const opentracing = require("opentracing");
+
+// go-swagger treats handles/expects arrays in the query string to be a string of comma joined values
+// so...do that thing. It's worth noting that this has lots of issues ("what if my values have commas in them?")
+// but that's an issue with go-swagger
+function serializeQueryString(data) {
+  if (Array.isArray(data)) {
+    return data.join(",");
+  }
+  return data;
+}
+
+module.exports = class WagPatch {
+
+  address;
+  timeout;
+
+  constructor(options) {
+    options = options || {};
+
+    if (options.discovery) {
+      this.address = discovery("wag-patch", "default").url();
+    } else if (options.address) {
+      this.address = options.address;
+    } else {
+      throw new Error("Cannot initialize wag-patch without discovery or address");
+    }
+    if (options.timeout) {
+      this.timeout = options.timeout
+    }
+  }
+
+  
+  wagpatch = (specialPatch, options, cb) => {
+    const params = {
+      
+        "specialPatch": specialPatch,
+      
+    };
+
+    if (!cb && typeof options === "function") {
+      cb = options;
+      options = undefined;
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    const timeout = options.timeout || this.timeout;
+    const span = options.span;
+
+    const headers = {
+
+    }
+
+    if (span) {
+      opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+      span.logEvent("PATCH /wagpatch");
+    }
+
+    const requestOptions = {
+      method: "PATCH",
+      uri: this.address + "/wagpatch",
+      json: true,
+      timeout,
+      headers,
+      body: {
+
+        "specialPatch": params.specialPatch,
+
+      },
```

Comment:
this doesn't seem like the correct request body--compare it to how the go client does it:

https://github.com/Clever/wag/blob/e085509847bc4607c9c82aebeb3be1ca6c1aef36/samples/gen-wag-patch/client/client.go#L107
+
https://github.com/Clever/wag/blob/e085509847bc4607c9c82aebeb3be1ca6c1aef36/samples/gen-wag-patch/models/patch_data.go#L15-L28